plugin/Plugin.pm

```diff
@@ -236,19 +236,23 @@ sub mixerVolumeCommand {
 	
 	my $members = $prefs->client($master)->get('members');	
 	return $originalVolumeHandler->($request) unless scalar @$members;
+		
+	# avoid recursing loop				
+	$master->_volumeDispatching(1);			
 
 	# be mindful that volume can be negative for mute	
 	my $oldVolume = abs($client->volume);
 	
-	# special handling of incremnent
+	# special handling of incremental vol changes
 	if ($newVolume =~ /^[\+\-]/) {
-		$newVolume += $oldVolume unless $client->volume < 0;
+		my $autoUnmute = $prefs->get('autoUnmute');
+		$newVolume += $oldVolume unless ($client->volume < 0 && !$autoUnmute);
 		$newVolume = 100 if $newVolume > 100;
 		$newVolume = 0 if $newVolume < 0;		
+		if ($client->volume < 0) {  # set the master volume if the group was muted
```

Comment:
It will re-enter the callback but will not technically recurse, as I moved the setting of the `_volumeDispatching`flag in front of it. However, it indeed **will** enter the `originalHandler`twice. I'll admit it was a lazy choice only taken because I wasn't sure how to modify the original parameter. It sounds as though `setParam` is the correct way to do it so I'll look into that. Another possibility that I considered would be to just return from the callback without invoking the `originalHandler ` when the original volume parameter is modified and let the second call do the actual work. In either case, I will honor your end-of-line preference for one-line conditional statements.

EDIT: I looked into `setParam` and can't find it. It looks like a `deleteParm `and `addParam` would be required. Is that what you had in mind?

Reply:
Done and tested using deleteParam/addParam.